compiler/rustc_index/src/vec.rs

```diff
@@ -138,12 +150,18 @@ macro_rules! newtype_index {
                 }
             }
 
+            /// Creates a new index from a given `u32`.
+            ///
+            /// # Safety
+            ///
+            /// This function is unsafe because it does not check for integer overflows.
```

Comment:
I don't think "integer overflow" is quite right here; integer overflow is not unsafe.

Reply:
Why *does* this have to be `unsafe`? 🤔

Reply:
Removing the `unsafe`:

```
    = note: initializing a layout restricted type's field with a value outside the valid range is undefined behavior
```

Reply:
I think I'll just copy that into the doc.

Reply:
I think because of this:

https://github.com/rust-lang/rust/blob/220ed09b26177ca4b6ab525f403d251024389a41/compiler/rustc_index/src/vec.rs#L108

Constructing a value outside the valid range would be UB.

Reply:
Thanks @camelid!